src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -224,11 +224,12 @@ protected override CompletionItemRules GetCompletionItemRules(ImmutableArray<Sym
                     item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
                 }
             }
-            else if (symbol.IsKind(SymbolKind.NamedType) || symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
+            else if (symbol is INamedTypeSymbol or IAliasSymbol { Target: INamedTypeSymbol })
             {
+                var namedTypeSymbol = symbol is INamedTypeSymbol ? (INamedTypeSymbol)symbol : (INamedTypeSymbol)((IAliasSymbol)symbol).Target;
                 // If this is a type symbol/alias symbol, also consider appending parenthesis when later, it is committed by using special characters,
                 // and the type is used as constructor
-                if (context.IsObjectCreationTypeContext)
+                if (context.IsObjectCreationTypeContext && CompletionUtilities.IsConstructorAccessibleAtPosition(completionContext.Position, namedTypeSymbol, context.SemanticModel))
```

Comment:
What if the constructor is accessible, but there is also a nested type?
In that case, we don't want `()` completion too, right?

For that, I already have an PR: https://github.com/dotnet/roslyn/pull/63015

Reply:
I just read your PR.
And to be honest I don't know which way is correct here. 🙂
I can potentially bring this to the design meeting